simulator fails with aiohttp < 3.9

<!--
Before opening a new issue, make sure you do the following:

- Check that your issue isn't already filed: https://github.com/pymodbus-dev/pymodbus/issues
- Check the discussions forum https://github.com/pymodbus-dev/pymodbus/discussions
- Prepare a short, runnable example that reproduce the issue with the latest development version of Pymodbus
-->
```
=========================== short test summary info ============================
FAILED test/sub_examples/test_examples.py::TestExamples::test_simulator - AttributeError: module 'aiohttp.web' has no attribute 'AppKey'
ERROR test/sub_server/test_simulator.py::TestSimulator::test_simulator_server_tcp - AttributeError: module 'aiohttp.web' has no attribute 'AppKey'
ERROR test/sub_server/test_simulator.py::TestSimulator::test_simulator_server_end_to_end - AttributeError: module 'aiohttp.web' has no attribute 'AppKey'
ERROR test/sub_server/test_simulator.py::TestSimulator::test_simulator_server_string - AttributeError: module 'aiohttp.web' has no attribute 'AppKey'
============ 1 failed, 1013 passed, 12 skipped, 3 errors in 39.91s =============
alex@antorak:~/git/pymodbus$ python3 -m pip list | grep aiohttp
aiohttp                            3.8.6
alex@antorak:~/git/pymodbus$ python3 --version
Python 3.10.12
```

```
< pyproject.toml >
simulator = [
    "aiohttp>=3.8.6;python_version<'3.12'",
    "aiohttp>=3.9.5;python_version=='3.12'"
]
```

Presumably some of the new simulator changes were done with `aiohttp==3.9.x`.  Either they need to be made to work with `3.8.x` or the required version must be bumped.

--- a/pymodbus/server/simulator/http_server.py
+++ b/pymodbus/server/simulator/http_server.py
@@ -218,7 +218,8 @@ class ModbusSimulatorServer:
         self.request_lookup = ServerDecoder.getFCdict()
         self.call_monitor = CallTypeMonitor()
         self.call_response = CallTypeResponse()
-        self.api_key: web.AppKey = web.AppKey("modbus_server")
+        app_key = getattr(web, 'AppKey', str)  # fall back to str for aiohttp < 3.9.0
+        self.api_key = app_key("modbus_server")
 
     async def start_modbus_server(self, app):
         """Start Modbus server as asyncio task."""

